### ISSUE TITLE
Extending Client Protocoll

### ISSUE THREAD

**Issue by mschaefer88** [eclipse/lsp4j#400]
I want to extend the client protocoll regarding lsp 3.15.0 and added the following method (for a first test):
```
public interface LanguageClient {
	(...)	
	@JsonRequest("window/workDoneProgress/create")
	default CompletableFuture<Void> workDoneProgressCreate(WorkDoneProgressCreateParams params) {
		throw new UnsupportedOperationException();
	}
}
```

But when accessing the method from my implementing server through the client interface I get the UnsupportedOperationException exception so I think I'm missing something cruical here.
Could you point me to the correct direction?

Thanks in advance!

**Comment by mschaefer88** [comment]
Solved it!
If anyone runs into the same issue:

The method name has to be the last segment of its endpoint address, ie. in this case **create**.

**Comment by jonahgraham** [comment]
Thanks for coming back with this. It needs to be documented properly.

I think the problem can also be fixed by using undocumented useSegment, e.g.:
```
@JsonRequest(value="window/workDoneProgress/create", useSegment=false)
	default CompletableFuture<Void> workDoneProgressCreate [...]
```

@mschaefer88 if the `useSegment=false` works for your original case I will fix the Javadoc for `JsonRequest` to make that clearer.

**Comment by mschaefer88** [comment]
@jonahgraham confirmed!

**Comment by jonahgraham** [comment]
Closed with #404
